modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/ClojureClientCodegen.java

```diff
@@ -171,9 +171,13 @@ public String getTypeDeclaration(Schema p) {
 
             return "(s/coll-of " + getTypeDeclaration(inner) + ")";
         } else if (ModelUtils.isMapSchema(p)) {
-            Schema inner = (Schema) p.getAdditionalProperties();
-
-            return "(s/map-of string? " + getTypeDeclaration(inner) + ")";
+            Object additionalProperties = p.getAdditionalProperties();
```

Comment:
i've filed https://github.com/OpenAPITools/openapi-generator/pull/18212 to rename the variable to just `ap` as the generator itself also has a property/field named "additionalProperties"